core/src/main/java/org/apache/accumulo/core/client/impl/TableOperationsImpl.java

```diff
@@ -1661,4 +1676,138 @@ public Locations locate(String tableName, Collection<Range> ranges) throws Accum
 
     return new LoctionsImpl(binnedRanges);
   }
+
+  @Override
+  public SummaryRetriever getSummaries(String tableName) {
+
+    return new SummaryRetriever() {
```

Comment:
Pull this out to a private static class instead of doing it inline? Would help testing.